### PR TITLE
[Add] Btn Component

### PR DESCRIPTION
**Pull request by ghost**
## What's this PR do ?

- Implement function A
- Fix bugs on function B

## Checklist

- [ ] Write/Update Unit Test (if have)
- [ ] Fill link PR into ticket and the opposite
- [ ] Validate UI/Model/API with latest specs
- [ ] Add UI pics to this PR (if have).

## Library

[flutter_screenutil: v5.8.4](https://pub.dev/packages/flutter_screenutil)

## Impacted Areas in Application

_(List features, api, models or services that this PR will affect)_

- foo/bar

### CODE REVIEW

File: lib/components/primary_button.dart [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+
+class PrimaryButton extends StatelessWidget {
+  const PrimaryButton(
+      {this.width,
+      this.height,
+      this.onPressed,
+      this.title,
+      this.textStyle,
+      super.key});
```
Comment (truyentd-2427) [inline lib/components/primary_button.dart]: add comma at last param then format code. It looks better
```suggestion
      super.key,});
```

File: lib/components/primary_button.dart [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      height: height ?? 50.h,
+      child: SizedBox(
+        width: width ?? size.width,
+        child: Text(
+          title ?? '',
+          textAlign: TextAlign.center,
+          style: TextStyle(
+              fontSize: 18.38.sp,
+              color: AppColors.white,
+              fontWeight: FontWeight.w600),
```
Comment (truyentd-2427) [inline lib/components/primary_button.dart]: same here
```suggestion
              fontWeight: FontWeight.w600,),
```

File: lib/components/primary_button.dart [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        ),
+      ),
+      onPressed: () {
+        onPressed?.call();
+      },
```
Comment (truyentd-2427) [inline lib/components/primary_button.dart]: just a suggestion, u can skip this
```suggestion
      onPressed: () => onPressed?.call(),
```

File: lib/components/primary_button.dart [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    this.title,
+    this.textStyle,
+    super.key,
+  });
+
+  final double? width;
+  final double? height;
+  final Function? onPressed;
+  final String? title;
+  final TextStyle? textStyle;
```
Comment (namnh-0652) [inline lib/components/primary_button.dart:21]: unused?

File: lib/components/primary_button.dart [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  Widget build(BuildContext context) {
+    return MaterialButton(
+      color: AppColors.crimsonApprox,
+      height: height ?? 50.h,
+      child: SizedBox(
+        width: width ?? 1.sw,
+        child: Text(
+          title ?? '',
+          textAlign: TextAlign.center,
+          style: TextStyle(
```
Comment (namnh-0652) [inline lib/components/primary_button.dart]: textStyle ?? default

File: lib/components/primary_button.dart [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+            fontSize: 18.38.sp,
+            color: AppColors.white,
+            fontWeight: FontWeight.w600,
+          ),
+        ),
+      ),
+      onPressed: () => onPressed?.call(),
```
Comment (namnh-0652) [inline lib/components/primary_button.dart]: `final Function() onPressed;`
then `onPressed: onPressed`

File: lib/components/secondary_button.dart [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  final Function? onPressed;
+  final String? title;
+  final double? width;
+  final double? height;
+  final TextStyle? textStyle;
+
+  @override
+  Widget build(BuildContext context) {
+    return OutlinedButton(
+      style: OutlinedButton.styleFrom(
```
Comment (namnh-0652) [inline lib/components/secondary_button.dart:29]: allow to custom this outline, `final OutlinedBorder? outlineBorder;`

**Review by namnh-0652 — APPROVED** [review]

**Pull request merged**